frontend/src/components/auth/PrivateRoute.tsx

```diff
@@ -16,12 +15,26 @@ const PrivateRoute: React.FC<PrivateRouteProps> = ({
   path,
 }: PrivateRouteProps) => {
   const { authenticatedUser } = useContext(AuthContext);
+  const location = useLocation();
+  const currentPath = location.pathname;
+
+  if (authenticatedUser === null) {
```

Comment:
nit: I think it might be better to check for _all_ possible falsey values here by simply checking for `if (!authenticatedUser)` for extra safety

Reply:
+1